server/src/main/kotlin/delta/codecharacter/server/daily_challenge/DailyChallengeService.kt

```diff
@@ -0,0 +1,45 @@
+package delta.codecharacter.server.daily_challenge
+
+import delta.codecharacter.dtos.DailyChallengeGetRequestDto
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.Period
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import java.util.Locale
+
+@Service
+class DailyChallengeService(
+    @Autowired private val dailyChallengeRepository: DailyChallengeRepository
+) {
+
+    @Value("\${environment.event-start-date}") val tempDate: String = ""
+
+    fun findNumberOfDays(): Int {
+        val formatter = DateTimeFormatter.ofPattern("yyyy/MM/dd/HH/mm/ss", Locale.ENGLISH)
+
+        val givenDateTime =
+            LocalDateTime.of(LocalDate.parse(tempDate, formatter), LocalTime.of(0, 0))
+                .atZone(ZoneId.of("Asia/Kolkata"))
+        val zdtNow = ZonedDateTime.now(ZoneId.of("Asia/Kolkata"))
+        val period: Period = Period.between(givenDateTime.toLocalDate(), zdtNow.toLocalDate())
+        return period.days
```

Comment:
- Change zone to ```GMT```
- To calculate the ```duration``` between two ```timestamps``` use ```Duration```.
- Get the start_date timestamp as ```YYYY-MM-DDTHH:MM:SSZ``` ( Eg : ```2023-01-08T15:07:43.376413035Z```
- The days calculation can be 
```kotlin
   val duration = Duration.between(instant1, instant2)
   /* instant1 = parsed instant of the start_date
      instant2 = current_instant
   */
   duration.toDays()
```